### ISSUE TITLE
Substitution of softmax function in torch?

### ISSUE THREAD

**Issue by tessavdheiden** [jxmorris12/categorical-vae#1]
Hi! 

Great repo and very well documented!

I was wondering, you have implemented your own [Gumbel Softmax function](https://github.com/jxmorris12/categorical-vae/blob/dcb24a12a094a5ddcc91ceb8a116e1d65f44ba37/models.py#L9), with gradient. 
Can you substitute it with the [Pytorch Gumbel Softmax function](https://pytorch.org/docs/stable/generated/torch.nn.functional.gumbel_softmax.html)?

Best,
Tessa

**Comment by jxmorris12** [comment]
@tessavdheiden Thanks! but why?

**Comment by tessavdheiden** [comment]
Because you can avoid the 35 (44-9) lines, with F.gumbel_softmax(x, dim=-1, tau=tau). So, it will make your code slimmer.

**Comment by jxmorris12** [comment]
Oh yeah, thanks. But I wanted to implement it myself, that was kind of the point.
